test/base_system_registration_dialog_test.rb

```diff
@@ -90,6 +90,23 @@
         end
       end
 
+      context "when user enters an invalid regcode" do
+        # include CRLF characters which are not allowed
+        let(:reg_code) { "\nmy-reg-code\r" }
+        it "displays error popup and does not register the system" do
+          expect(Yast::UI).to receive(:QueryWidget).with(:reg_code, :Value)
```

Comment:
np: IMHO, you are not testing this, so I would use `allow` here to make it clear. The same for the next two lines.